catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/QueryApplication.java

```diff
@@ -208,4 +225,48 @@ public void setFeatureService(FeatureService featureService) {
   public void setEndpointUtil(EndpointUtil util) {
     this.util = util;
   }
+
+  private Map<String, Object> getLatLonLiteral(String query) {
+    String queryStringNumbersOnly = query.replaceAll("[^\\.\\-\\d]", " ");
+    if (StringUtils.isBlank(queryStringNumbersOnly)) {
+      return null;
+    }
+    List<String> numbersAsStrings =
+        new ArrayList<>(Arrays.asList(queryStringNumbersOnly.trim().split("\\s+")));
+    if (numbersAsStrings.size() % 2 != 0) {
+      numbersAsStrings.remove(numbersAsStrings.size() - 1);
+      if (numbersAsStrings.isEmpty()) {
+        return null;
+      }
+    }
+    List<Double> numbers =
+        numbersAsStrings
+            .stream()
+            .map(number -> Double.parseDouble(number))
+            .collect(Collectors.toCollection(ArrayList::new));
+    List<ArrayList<Double>> latLons = getLatLons(numbers);
+    Map<String, Object> latLonList =
+        new HashMap() {
+          {
+            put("id", "LITERAL");
+            put("name", "Use detected Lat/Lon");
+            put("geo", latLons);
+          }
+        };
+    return latLonList;
+  }
+
+  private List<ArrayList<Double>> getLatLons(List<Double> numbers) {
+    List<ArrayList<Double>> latLons = new ArrayList<>();
+    for (int index = 0; index < numbers.size(); index += 2) {
+      double latitude = numbers.get(index);
+      if (latitude >= -90 && latitude <= 90) {
+        double longitude = numbers.get(index + 1);
+        if (longitude >= -180 && longitude <= 180) {
+          latLons.add(new ArrayList<>(Arrays.asList(latitude, longitude)));
```

Comment:
If you change every declaration to `List` instead of `ArrayList` you don't have to wrap every `Arrays.asList` in your code with `new ArrayList<>(...`